File: app/src/main/java/me/ykrank/s1next/view/activity/EditPostActivity.java
package me.ykrank.s1next.view.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.v4.app.Fragment;
import android.support.v4.app.FragmentManager;

import me.ykrank.s1next.R;
import me.ykrank.s1next.data.api.model.Post;
import me.ykrank.s1next.data.api.model.Thread;
import me.ykrank.s1next.view.fragment.EditPostFragment;

/**
 * An Activity to new a thread.
 */
public final class EditPostActivity extends BaseActivity {

    private static final String ARG_THREAD = "thread";
    private static final String ARG_POST = "post";

    private EditPostFragment mFragment;

    public static void startActivityForResultMessage(Fragment fragment, int requestCode, Thread thread, Post post) {
        Intent intent = new Intent(fragment.getContext(), EditPostActivity.class);
        intent.putExtra(ARG_THREAD, thread);
        intent.putExtra(ARG_POST, post);
        fragment.startActivityForResult(intent, requestCode);
    }

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_base_without_drawer_and_scrolling_effect);

        setupNavCrossIcon();

        Intent intent = getIntent();
        Thread mThread = intent.getParcelableExtra(ARG_THREAD);
        Post mPost = intent.getParcelableExtra(ARG_POST);
        setTitle(R.string.title_new_thread);

        FragmentManager fragmentManager = getSupportFragmentManager();
        Fragment fragment = fragmentManager.findFragmentByTag(EditPostFragment.Companion.getTAG());
        if (fragment == null) {
            mFragment = EditPostFragment.Companion.newInstance(mThread, mPost);
            fragmentManager.beginTransaction().add(R.id.frame_layout, mFragment,
                    EditPostFragment.Companion.getTAG()).commit();
        } else {
            mFragment = (EditPostFragment) fragment;
        }
    }

    /**
     * Show {@link android.app.AlertDialog} when reply content is not empty.
     */
    @Override
    public void onBackPressed() {
        if (mFragment.isEmoticonKeyboardShowing()) {
            mFragment.hideEmoticonKeyboard();
        } else {
            super.onBackPressed();
        }
    }
}

File: app/src/main/java/me/ykrank/s1next/view/activity/EditPostActivity.kt
package me.ykrank.s1next.view.activity

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.Fragment
import me.ykrank.s1next.R
import me.ykrank.s1next.data.api.model.Post
import me.ykrank.s1next.data.api.model.Thread
import me.ykrank.s1next.view.fragment.EditPostFragment

/**
 * An Activity to edit post.
 */
class EditPostActivity : BaseActivity() {

    private lateinit var mFragment: EditPostFragment

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_base_with_ime_panel)

        setupNavCrossIcon()

        val intent = intent
        val mThread = intent.getParcelableExtra<Thread>(ARG_THREAD)
        val mPost = intent.getParcelableExtra<Post>(ARG_POST)

        val fragmentManager = supportFragmentManager
        val fragment = fragmentManager.findFragmentByTag(EditPostFragment.TAG)
        if (fragment == null) {
            mFragment = EditPostFragment.newInstance(mThread, mPost)
            fragmentManager.beginTransaction().add(R.id.frame_layout, mFragment,
                    EditPostFragment.TAG).commit()
        } else {
            mFragment = fragment as EditPostFragment
        }
    }

    /**
     * Show [android.app.AlertDialog] when reply content is not empty.
     */
    override fun onBackPressed() {
        if (mFragment.isToolsKeyboardShowing) {
            mFragment.hideToolsKeyboard()
        } else {
            super.onBackPressed()
        }
    }

    companion object {

        private val ARG_THREAD = "thread"
        private val ARG_POST = "post"

        fun startActivityForResultMessage(fragment: Fragment, requestCode: Int, thread: Thread, post: Post) {
            val intent = Intent(fragment.context, EditPostActivity::class.java)
            intent.putExtra(ARG_THREAD, thread)
            intent.putExtra(ARG_POST, post)
            fragment.startActivityForResult(intent, requestCode)
        }
    }
}
